core/iwasm/aot/arch/aot_reloc_riscv.c

```diff
@@ -170,14 +170,32 @@ rv_get_val(uint16 *addr)
     return ret;
 }
 
-/* Set a val to given address */
+/**
+ * Set a val to given address
+ *
+ * Zifencei extension is moved out from I extension in RISC-V spec 20191213
+ * (rv32i/rv64i version 2.1 and later), new version of GNU toolchain or clang
+ * may not support it if no explicit 'zifencei' is specified in arch attribute.
+ *
+ * Refer to:
+ * https://github.com/riscv-non-isa/riscv-c-api-doc/blob/master/riscv-c-api.md
+ */
+#if __riscv_i >= 2001000
+__attribute__((target("arch=+zifencei")))
+#endif
 static void
 rv_set_val(uint16 *addr, uint32 val)
 {
     *addr = (val & 0xffff);
     *(addr + 1) = (val >> 16);
 
+    /* REVISIT: Zephyr's GCC toolchain does not support attribute
+     * target("arch=+zifencei") and on MCU fence.i may equal to nop, so we can
+     * disable it here.
+     */
```

Comment:
I believe Zephyr's toolchain does support `zifencei`, but some targets do not support the extension. Have you found that not to be the case?

Reply:
I mean zephyr's toolchain should support zifencei extension but don't support enable it by `__attribute__((target("arch=+zifencei")))`, consider as an RTOS, the fence.i on micro controller may not actually sense, so can be ignored here.

Reply:
I believe just some targets don't support it (such as `esp32c3-devkitm`). However, other Zephyr targets do support it (example: https://github.com/zephyrproject-rtos/zephyr/blob/23cf38934c0f68861e403b22bc3dd0ce6efbfa39/soc/riscv/riscv-privileged/sifive-freedom/Kconfig.soc#L18).

Reply:
Thanks, does `RISCV_ISA_EXT_ZIFENCEI` visible for us? How can we check it?

Reply:
It is a Kconfig option, so it would be available as `CONFIG_RISCV_ISA_EXT_ZIFENCEI` (e.g. https://github.com/zephyrproject-rtos/zephyr/blob/23cf38934c0f68861e403b22bc3dd0ce6efbfa39/cmake/compiler/gcc/target_riscv.cmake#L48). Let me see if there is a more portable option here though 👍🏻

Reply:
If so, let's check this option first.